src/app/views/user/user-settings/user-profile/user-profile.component.html

```diff
@@ -174,17 +174,17 @@
 
         <div class="profile-input-table-container">
           <div class="profile-input-table-row-container" *ngFor="let userLanguage of profileForm.value.user_languages">
-            <div class="profile-input-table-row-inner-container" *ngIf="userLanguage.proficiency && userLanguage.proficiency.proficiency !== 5">
+            <div class="profile-input-table-row-inner-container" *ngIf="userLanguage.proficiency">
               <div class="profile-input-table-row-inner-item-container">
                 <div class="profile-input-table-row-inner-item-text">
                   {{userLanguage.language.name}}
                 </div>
-                <div class="profile-input-table-row-inner-icon-container" (click)="onRemoveUserLanguage(userLanguage)">
+                <div class="profile-input-table-row-inner-icon-container" (click)="onRemoveUserLanguage(userLanguage)" *ngIf="userLanguage.proficiency.proficiency !== 5">
                   <i class="fa fa-times" aria-hidden="true"></i>
                 </div>
               </div>
               <div class="profile-input-table-row-inner-dropdown-container">
-                <autocomplete-dropdown [lookupType]="'list'" [enumList]="languageProficiencyLevelsAvailable" [(destination)]="userLanguage.proficiency" [placeholder]="'user.profile.form.language.proficiency.placeholder' | translate" [autocompleteResultLabelFunction]="namePropertyLabel"
+                <autocomplete-dropdown [lookupType]="'list'" [isDisabled]="userLanguage.proficiency.proficiency === 5" [enumList]="languageProficiencyLevelsAvailable" [(destination)]="userLanguage.proficiency" [placeholder]="'user.profile.form.language.proficiency.placeholder' | translate" [autocompleteResultLabelFunction]="namePropertyLabel"
```

Comment:
But the magic number should be changed (see enums.ts for the constant) and for readability spanning multiple lines might be a good consideration for refactoring throughout the html codebase. Not many html elements are that long so shouldnt be much of hassle.

enums.ts also need a refactoring since Boris made it strangely mimic enums with the use of lists and not dictionary object structure. But use the list to retrieve the constant for now and I will refactor the enums.ts to appropriate files and structure later.